src/main/java/stsjorbsmod/cards/cull/OldBook.java

```diff
@@ -38,13 +42,34 @@ public OldBook() {
         tags.add(LEGENDARY);
     }
 
+    /**
+     * used in ShowCardAndAddToHandEffect::new which gets used in Discovery, Card Potions, and MakeTempCardInHandAction (Dead Branch and console command HandAdd)
+     */
+    @Override
+    public void triggerWhenCopied() {
+        super.triggerWhenCopied();
+        priority = currentPriority.incrementAndGet();
+    }
+
+    /**
+     * used in FastCardObtainEffect::update (combat reward cards) and ShowCardAndObtainEffect::update (just about every other card acquisition method events/transforms/etc).
+     * overloading the meaning of shrink() because lazy.
+     */
+    @Override
+    public void shrink() {
```

Comment:
I don't think this is the method you want to override as this gets called anytime a card moves from the hand to the discard pile during combat. 

In fact, I think the solution is simpler: The `masterDeck` CardGroup object holds  all cards obtained in an `ArrayList`, and the order is unaffected anywhere in the base game. We should be able to check the master deck for any `DeathPrevention` cards, and the order of the cards in that list should work great for the order in which they're obtained. Checking cards added by potions/nilrys/dead branch etc could then be checked if none are found in the `masterDeck` using your solution here. 

I think this might make the `DeathPreventionPatch` below simpler as well, as an added benefit.

Reply:
Good catch on shrink() being called on discard! I wonder if it's part of the animation of the card flying to a pile (draw, discard, deck)...

I'm not sure that checking the master deck is necessarily simpler. Shrieking Hat needs to be in hand in order to prevent death not just existing somewhere.

Reply:
Oh i'm dumb, we already know that. Though I still don't think it's simpler.